data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/AcknowledgementSetMonitor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.acknowledgements;
+
+import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.event.DefaultEventHandle;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.Set;
+import java.util.HashSet;
+
+public class AcknowledgementSetMonitor implements Runnable {
+    private final Set<AcknowledgementSet> acknowledgementSets;
+    private final ReentrantLock lock;
+
+    private DefaultAcknowledgementSet getAcknowledgementSet(final EventHandle eventHandle) {
+        return (DefaultAcknowledgementSet)((DefaultEventHandle)eventHandle).getAcknowledgementSet();
```

Comment:
Why do we need to cast here ? Can getAcknowledgementSet be method of the interface ?

Reply:
it is a private method and need not be exposed externally. Secondly, we may change this implementation in the future if needed.